lib/spack/spack/test/config.py

```diff
@@ -791,6 +791,15 @@ def test_single_file_scope_section_override(tmpdir, config):
             '/x/y/z', '$spack/var/spack/repos/builtin']
 
 
+def test_write_empty_single_file_scope(tmpdir):
+    env_schema = spack.schema.env.schema
+    scope = spack.config.SingleFileScope(
+        'test', str(tmpdir.ensure('config.yaml')), env_schema, ['spack'])
+    scope.write_section('config')
+    # confirm we can write empty config
+    assert scope.get_section('config') is None
```

Comment:
This test is specifically to cover the case in which the data is `None`, because the `SingleConfigScope` was instantiated with no data. If the data isn't still `None` when we read it back, something went wrong (in the code that specifically returns `None`). This is a test that was added for coverage in some corner-case code that I don't think can possibly be reached by a user who is not using `spack-python` to do awful things to the configs, but I figured it should still be robust to that.